### PR TITLE
166187832 user should be able to search articles

### PR DESCRIPTION
**Pull request by alexxsanya**
#### What does this PR do?

user should be able to search articles 

#### Description of Task to be completed?

- enable search form to trigger database query
- retrieve and display search results.
- add tests for the search components

#### How should this be manually tested?

- Git fetch this branch `ft-search-functionality-166187832`
- Run the application with `npm start`.
- Or visit the review app [here](https://authors-frontend-staging-pr-21.herokuapp.com/)
- Enter a value to search for in the search field and click search

#### What are the relevant pivotal tracker stories?

[#166187832](https://www.pivotaltracker.com/story/show/#166187832)

### CODE REVIEW

File: src/Mainstore/Store.js [hunk 1]
```diff
@@ -6,14 +6,14 @@ import LatestArticleReducer from '../Landing/redux/reducers/latestArticlesReduce
...
-
+import SearchArticleReducer from '../Articles/redux/reducers/SearchArticleReducer';
 
 const reducers = combineReducers({
   latest_articles: LatestArticleReducer,
   login: loginReducer,
   signUpUser: signUpReducer,
   allArticleReducer,
-
+  searched_articles: SearchArticleReducer,
```
Comment (neelxie) [inline src/Mainstore/Store.js]: @alexxsanya please change and use the camelcase convention as expected of us for this particular project.

File: src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  status: true,
+  is_fetching: true,
+};
+
+const searchFailedAction = {
+  type: SEARCH_ARTICLE_FAILED,
+  message: undefined,
+  items: [],
+  status: false,
+  is_fetching: false,
```
Comment (neelxie) [inline src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx]: @alexxsanya please do change all incidents of this in your task.

File: src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+const searchAction = {
+  type: SEARCH_ARTICLE_SUCCESS,
+  payload: [{ articles: ['1'] }],
+  message: 'Articles successfully retrieved',
+  status: true,
+  is_fetching: true,
+};
+
+const searchFailedAction = {
+  type: SEARCH_ARTICLE_FAILED,
```
Comment (neelxie) [inline src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx:15]: @alexxsanya This as well, please.

**Review by neelxie — CHANGES_REQUESTED** [review]
@alexxsanya Please take note of the conventions for all the naming cases, and try to desist from mixing python conventions with javascript conventions.

File: src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx (hunk 2 above)
Comment (alexxsanya) [reply src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx]: @neelxie thanks for this feedback. I am working on it.

File: src/Mainstore/Store.js (hunk 1 above)
Comment (alexxsanya) [reply src/Mainstore/Store.js]: @neelxie this has been fixed.

File: src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx (hunk 2 above)
Comment (alexxsanya) [reply src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx]: @neelxie I am done working on this.

File: src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx (hunk 3 above)
Comment (alexxsanya) [reply src/Articles/tests/redux/reducers/SearchArticlesReducers.test.jsx:15]: @this has been fixed as well

**Review by neelxie — CHANGES_REQUESTED** [review]
@alexxsanya good work but looking at the wording of your error message below, I doubt if my mother was using this website she would understand the context of the word 'key' I believe only developers would understand that, I would recommend to changing that particular phrase to something more appropriate to ordinary users, try emulating what google usually responds with in such circumstances. And good work.

Screenshot 2019-07-24 at 09 47 55

this is what you get when you input spaces

Screenshot 2019-07-20 at 15 15 04

When you input this below

Screenshot 2019-07-24 at 10 10 03

here is the response 

Screenshot 2019-07-24 at 10 10 15

**Comment by alexxsanya** [comment]
> @alexxsanya good work but looking at the wording of your error message below, I doubt if my mother was using this website she would understand the context of the word 'key' I believe only developers would understand that, I would recommend to changing that particular phrase to something more appropriate to ordinary users, try emulating what google usually responds with in such circumstances. And good work.
> 
> Screenshot 2019-07-24 at 09 47 55

@neelxie I doublt I can achieve this scenario with Google

> @alexxsanya good work but looking at the wording of your error message below, I doubt if my mother was using this website she would understand the context of the word 'key' I believe only developers would understand that, I would recommend to changing that particular phrase to something more appropriate to ordinary users, try emulating what google usually responds with in such circumstances. And good work.
> 
> Screenshot 2019-07-24 at 09 47 55

@neelxie thanks for pointing this out. I have updated the phrases accordingly.
Screenshot 2019-07-24 at 10 01 14

**Comment by alexxsanya** [comment]
> @alexxsanya good work but looking at the wording of your error message below, I doubt if my mother was using this website she would understand the context of the word 'key' I believe only developers would understand that, I would recommend to changing that particular phrase to something more appropriate to ordinary users, try emulating what google usually responds with in such circumstances. And good work.
> 
> Screenshot 2019-07-24 at 09 47 55
> 
> this is what you get when you input spaces
> 
> Screenshot 2019-07-20 at 15 15 04
> 
> When you input this below
> 
> Screenshot 2019-07-24 at 10 10 03
> 
> here is the response
> 
> Screenshot 2019-07-24 at 10 10 15

@neelxie You are getting that response because that data is changing the acceptable defined routes hence getting a blank screen. Also, be assured that when a user enters an invalid word with special characters likes those don't hit the search endpoint so there is no possibility of compromising the database.

**Review by neelxie — APPROVED** [review]
@alexxsanya good articulate stuff here once again good work brother. Keep up the great work.

**Review by HabibSentongo — APPROVED** [review]
The functionality is working well and you have covered it with tests. Good work here.

**Review by jennizalwango — COMMENTED** [review]
Screenshot 2019-07-24 at 12 58 07 PM
thanks @alexxsanya  l have  reviewed your work this seems to be working well on my side

**Review by Emmanuelaaron — APPROVED** [review]
Great work @alexxsanya ,
everything seems to be working perfectly on my end.

**Review by jennizalwango — APPROVED** [review]
@alexxsanya  since you have worked on the feedback and l tested your thing out l think your good to go.

**Pull request merged**